#!/usr/bin/env python3
PACKAGE = "ps_planner"
import rospkg
import yaml, os
from dynamic_reconfigure.parameter_generator_catkin import *

ros_path = rospkg.RosPack().get_path('ps_planner')


gen = ParameterGenerator()
gen.add("evasion_dist", double_t, 0, "Orthogonal distance of the apex to the obstacle", 0.35, 0.0, 1.25)
gen.add("obs_traj_tresh", double_t, 0, "Threshold of the obstacle towards raceline to be considered for evasion", 1.5, 0.1, 2)
gen.add("spline_bound_mindist", double_t, 0, "Splines may never be closer to the track bounds than this param in meters", 0.20, 0.05, 1.0)
gen.add("lookahead_dist", double_t, 0, "Lookahead distance in meters", 30.0, 1.0, 50.0)
gen.add("avoidance_resolution", int_t, 0, "Number of points used to generate avoidance path", 30, 10, 100)
gen.add("back_to_raceline_before", double_t, 0, "Distance in meters before obstacle to stay on the raceline", 6, 0.5, 10.0)
gen.add("back_to_raceline_after", double_t, 0, "Distance in meters after obstacle to go back on the raceline", 8, 0.5, 10.0)

gen.add("avoid_static_obs", bool_t, 0, "Avoid static obstacles", False)

exit(gen.generate(PACKAGE, "sqp_dynamic_tuner", "dyn_sqp_tuner"))